src/nsrt_learning.py

```diff
@@ -146,118 +149,135 @@ def segment_trajectory(trajectory: GroundAtomTrajectory) -> List[Segment]:
     return segments
 
 
-def learn_strips_operators(segments: Sequence[Segment], verbose: bool = True,
-        ) -> Tuple[List[STRIPSOperator], List[Partition]]:
-    """Learn operators given the segmented transitions.
-    """
+def _get_initial_nids(segments: Sequence[Segment], verbose: bool
+                      ) -> List[_NSRTIntermediateData]:
     # Partition the segments according to common effects.
-    params: List[Sequence[Variable]] = []
-    parameterized_options: List[ParameterizedOption] = []
-    option_vars: List[Tuple[Variable, ...]] = []
-    add_effects: List[Set[LiftedAtom]] = []
-    delete_effects: List[Set[LiftedAtom]] = []
-    partitions: List[Partition] = []
+    nids = []
     for segment in segments:
-        if segment.has_option():
+        if not CFG.do_option_learning:
             segment_option = segment.get_option()
             segment_param_option = segment_option.parent
             segment_option_objs = tuple(segment_option.objects)
         else:
             segment_param_option = DummyOption.parent
             segment_option_objs = tuple()
-        for i in range(len(partitions)):
+        for nid in nids:
             # Try to unify this transition with existing effects.
             # Note that both add and delete effects must unify,
             # and also the objects that are arguments to the options.
-            part_param_option = parameterized_options[i]
-            part_option_vars = option_vars[i]
-            part_add_effects = add_effects[i]
-            part_delete_effects = delete_effects[i]
+            (nid_param_option, nid_option_vars) = nid.option_spec
+            nid_add_effects = nid.op.add_effects
```

Comment:
can we just plug these in directly, like we do for segment.add_effects etc, rather than making new variables?